### PR TITLE
Renaming and tests integration v2 events

### PR DESCRIPTION
**Pull request by mariomac**
#### Description of the changes

* Removed redundant pointers.
* Removed ugly "Protocol2" suffixes.
* Added tests to check that the multi-entity events work as expected.
* Deprecated `v1` integration.

** Please notice ** : base for this PR is branch `mmacias/sdkv1v2`. As soon as the mentioned branch is merged, please change the base of this PR to `master`.

### Author

- [ ] add a risk label after carefully considering the "blast radius" of your changes
- [ ] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [ ] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: docs/tutorial-code/redis.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func main() {
 
 	if args.All || args.Metrics {
 		ms := integration.NewMetricSet("RedisSample")
-		fatalIfErr(populateMetrics(ms))
+		fatalIfErr(populateMetrics(&ms))
```
Comment (MagdaPaj) [inline docs/tutorial-code/redis.go]: If this has been changed, then also tutorial documentation should be. It appears there couple of times. https://github.com/newrelic/infra-integrations-sdk/blob/master/docs/tutorial.md

File: sdk/v1/data.go [hunk 2]
```diff
@@ -68,8 +70,8 @@ func NewIntegration(name string, version string, arguments interface{}) (*Integr
 		ProtocolVersion:    "1",
 		IntegrationVersion: version,
 		Inventory:          make(Inventory),
-		Metrics:            make([]*metric.MetricSet, 0),
-		Events:             make([]*Event, 0),
+		Metrics:            make([]metric.MetricSet, 0),
```
Comment (MagdaPaj) [inline sdk/v1/data.go]: I don't know if we should do the breaking change in protocol 1, if we're going to deprecate it.

Comment (MagdaPaj) [reply sdk/v1/data.go]: But well, the breaking change was already done for events. So maybe we shouldn't care.

File: sdk/v1/data_test.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 	"testing"
 
 	sdk_args "github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/stretchr/testify/assert"
```
Comment (MagdaPaj) [inline sdk/v1/data_test.go]: the same: https://github.com/newrelic/infra-integrations-sdk/pull/50#discussion_r181027556

File: sdk/v1/data.go (hunk 2 above)
Comment (mariomac) [reply sdk/v1/data.go]: Let's discuss it later in meetings.

**Comment by mariomac** [comment]
Ok jenkins

**Comment by mariomac** [comment]
ok jenkins

**Pull request merged**